### PR TITLE
fix(tooltip): tooltip sometimes not hidden after element is disabled.

### PR DESCRIPTION
**Pull request by SanderElias**
fix(tooltip): tooltip sometimes not hidden after element is disabled.

use an mutationobserver to make sure the tooltip is hidden after an element becomes disabled.

fixes: #2602

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
@jelbourn, @rschmukler  - can you review please.

**Comment by ThomasBurleson** [comment]
@SanderElias - please rebase.

File: src/components/tooltip/tooltip.js [hunk 1]
```diff
@@ -128,6 +128,19 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
 
       var ngWindow = angular.element($window);
 
+      if (parent[0] && 'MutationObserver' in $window) {
```
Comment (jelbourn) [inline src/components/tooltip/tooltip.js]: This block definitely needs some comments explaining what's going on.

Comment (jelbourn) [inline src/components/tooltip/tooltip.js]: Can you add a unit test that covers this behavior?

File: src/components/tooltip/tooltip.js [hunk 2]
```diff
@@ -128,6 +128,19 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
...
+        // use an mutationObserver to tackle #2602
+        var observer = new MutationObserver(function(mutations) {
```
Comment (jelbourn) [inline src/components/tooltip/tooltip.js]: If `observer` is going to be referenced later in the `"$destroy"` handler, it should have a more specific name.

File: src/components/tooltip/tooltip.js [hunk 3]
```diff
@@ -128,6 +128,19 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
...
+          mutations
+            .filter(function (m) { return m.attributeName === 'disabled'; })
+            .map(function (m) { if (parent[0].disabled) { setVisible(false); } });
+        });
+
+        observer.observe(parent[0], { attributes: true});
+      };
+
+      //------------------------------------------
```
Comment (jelbourn) [inline src/components/tooltip/tooltip.js]: Remove separator comment line.

File: src/components/tooltip/tooltip.js [hunk 4]
```diff
@@ -140,6 +153,8 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
       scope.$on('$destroy', function() {
         ngWindow.off('blur', windowBlurHandler);
         ngWindow.off('scroll', windowScrollHandler);
+        // remove the observer if its there
```
Comment (jelbourn) [inline src/components/tooltip/tooltip.js]: This comment doesn't tell you anything.

**Comment by SanderElias** [comment]
This PR is bases on a wrong branch. I'm going to close and delete it, and put in a fresh copy.

**Pull request closed without merging**